evm_trace/gas.py

```diff
@@ -0,0 +1,54 @@
+from typing import Dict, List
+
+from evm_trace.base import CallTreeNode
+
+Address = str
+Calldata = bytes
+GasUsage = int
+
+GasReport = Dict[Address, Dict[Calldata, List[GasUsage]]]
+
+
+def extract_gas_report(calltree: CallTreeNode) -> GasReport:
+    """
+    Extracts a gas report object from a :class:`~evm_trace.base.CallTreeNode`
+
+    Args:
+        calltree (:class:`~evm_trace.base.CallTreeNode`): extractable call tree
+
+    Returns:
+        :class:`~evm_trace.gas.GasReport`: Gas report structure from a call tree.
+    """
+    report = {
+        calltree.address: {calltree.calldata: [calltree.gas_cost] if calltree.gas_cost else []}
+    }
+
+    for call in calltree.calls:
+        report = combine_gas_reports(report, extract_gas_report(call))
+
+    return report
+
+
+def combine_gas_reports(report1: GasReport, report2: GasReport) -> GasReport:
```

Comment:
Is this method intended for external usage? If yes, it may be awkward to combine a parent and its children because you have to always manage the same loop (like the method get gas report method is doing)

I recommend one of the following:

1. Make the method accept a parent (singular) and children (list)
2. Make the method a non-public helper method.

Reply:
Hmm, maybe make it look more like this: `merge_reports(*reports: List[GasReport]) -> GasReport`